### PR TITLE
docs(cli): 💅 add option grouping and examples

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR improves the help command. 💅

You can print the help text to the console by running `code-pushup <any command or none> --help` or `code-pushup help`.

**Included changes:**  
- adoption of cli wrapper. Added `groups` and `examples` properties and logic
- added groups to cli object
- added examples to cli object

**Terminal output:**  

| before improvements | after improvements✨ |
| -- | -- |
| Screenshot 2024-03-09 at 03 56 44  | Screenshot 2024-03-09 at 03 48 43 |

### CODE REVIEW

**Review by matejchalk — COMMENTED** [review]
I think this is a nice improvement :slightly_smiling_face:

File: e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap [hunk 1]
```diff
@@ -11,26 +11,35 @@ Commands:
...
+Upload Options:
       --upload.organization  Organization slug from portal              [string]
       --upload.project       Project slug from portal                   [string]
       --upload.server        URL to your portal server                  [string]
       --upload.apiKey        API key for the portal server              [string]
-      --onlyPlugins          List of plugins to run. If not set all plugins are
-                             run.                          [array] [default: []]
-  -h, --help                 Show help                                 [boolean]
+
+Options:
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: That is the options of the command only. It should be gone in upload... 

Not sure how I can change that without adding groups to everywhere command.

[the remark this answers is not in the document]
Comment (matejchalk) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: How about adding it like this? :thinking: 

```ts
export const groups = {
  'Global Options:': Object.keys(yargsGlobalOptionsDefinition()),
  'Persist Options:': Object.keys(yargsPersistConfigOptionsDefinition()),
  'Upload Options:': Object.keys(yargsUploadConfigOptionsDefinition()),
  // add here ⬇️
  'Other Options': Object.keys(yargsOnlyPluginsOptionsDefinition()),
};
```

[the remark this answers is not in the document]
Comment (BioPhoton) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: What about per command?

**Code:**  
Screenshot 2024-03-11 at 15 29 03

**Prints:**  
Screenshot 2024-03-11 at 15 28 45

[the remark this answers is not in the document]
Comment (matejchalk) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: Could be :man_shrugging:

[the remark this answers is not in the document]
Comment (matejchalk) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: But I'd also be fine with having `--onlyPlugins` under Global or Other. We don't have to over-engineer it while we only have this one command-specific option.

[the remark this answers is not in the document]
Comment (BioPhoton) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: Ill merge it into global...

[the remark this answers is not in the document]
Comment (BioPhoton) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:34]: done!

File: e2e/cli-e2e/tests/help.e2e.test.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ describe('CLI help', () => {
     });
     const helpCommandResult = await executeProcess({
       command: 'code-pushup',
-      args: ['--help'],
+      args: ['help'],
```
Comment (matejchalk) [inline e2e/cli-e2e/tests/help.e2e.test.ts:21]: The point of this test is to check that `--help` has the same output as `help`. But now you're comparing `help` with `help` :sweat_smile:

File: e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap [hunk 3]
```diff
@@ -11,26 +11,88 @@ Commands:
...
+
+Examples:
+  code-pushup collect --config tsconfig.te  Use the config from \`tsconfig.test.j
+  st.json                                   son\` and only collect audits
+  code-pushup print-config --onlyPlugins    print the config object but from the
+                                             plugins only list given slugs
+"
+`;
+
+exports[`CLI help > should print help with command help 1`] = `
```
Comment (matejchalk) [inline e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:47]: The `CLI help > should print help with argument ---help 1` snapshot above is now obsolete, so should be deleted.

**Review by matejchalk — CHANGES_REQUESTED** [review]

File: e2e/cli-e2e/tests/help.e2e.test.ts (hunk 2 above)
Comment (BioPhoton) [reply e2e/cli-e2e/tests/help.e2e.test.ts]: reverted

File: e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap (hunk 3 above)
Comment (BioPhoton) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap]: added test instead of deleting

**Review by matejchalk — APPROVED** [review]
Looks good to me :+1: 

Do you also want to integrate the `compare` options from #556, as discussed here? https://github.com/code-pushup/cli/pull/556#discussion_r1520321177

**Pull request merged**